test/PizzeriaTest.cpp

```diff
@@ -52,16 +56,52 @@ TEST_F(PizzeriaTest, completeOrderWithStubPizza)
 TEST_F(PizzeriaTest, calculatePriceForPizzaMock)
 {   
     // Given
-    PizzaMock* mock = new PizzaMock{};
+    NaggyMock<PizzaMock>* mock = new NaggyMock<PizzaMock>{};
     Pizzas pizzas = {mock};
+    EXPECT_CALL(*mock, getName()).WillOnce(Return("naggy_mock"));
     EXPECT_CALL(*mock, getPrice()).WillOnce(Return(40.0));
-    
+    EXPECT_CALL(*mock, getBakingTime).Times(1);
+    EXPECT_CALL(mockTimer, sleepFor).Times(1);
+
     // When
     auto orderId = pizzeria.makeOrder(pizzas);
+    pizzeria.bakePizzas(orderId);
     auto price = pizzeria.calculatePrice(orderId);
 
     // Then
     ASSERT_EQ(40, price);
 
     delete mock;
 }
+
+TEST_F(PizzeriaTest, orderOneStubAndTwoMocks) {
+    // Given
+    PizzaStub* stubPizza = new PizzaStub{"stub"};
+    StrictMock<PizzaMock>* strictMockPizza = new StrictMock<PizzaMock>{};
+    NiceMock<PizzaMock>* niceMockPizza = new NiceMock<PizzaMock>{};
+
+    Pizzas pizzas = {stubPizza, strictMockPizza, niceMockPizza};
+
+    EXPECT_CALL(*strictMockPizza, getName()).WillOnce(Return("scrict_mock"));
+    EXPECT_CALL(*strictMockPizza, getBakingTime()).WillOnce(Return(minutes(1)));
+    EXPECT_CALL(*strictMockPizza, getPrice()).WillOnce(Return(30.0));
+
+    EXPECT_CALL(*niceMockPizza, getName()).WillOnce(Return("nice_mock"));
+    EXPECT_CALL(*niceMockPizza, getBakingTime()).WillOnce(Return(minutes(1)));
+    EXPECT_CALL(*niceMockPizza, getPrice()).WillOnce(Return(35.0));
+
+    EXPECT_CALL(mockTimer, sleepFor).Times(3);
```

Comment:
you should expect 3 calls, but each with different parameter - baking time of each pizza.

Reply:
Indeed. Add three calls, each with different parameter.